wagtail/api/v2/views.py

```diff
@@ -508,8 +512,29 @@ def get_base_queryset(self):
         for restricted_page in restricted_pages:
             queryset = queryset.not_descendant_of(restricted_page, inclusive=True)
 
-        # Filter by site
-        site = Site.find_for_request(self.request)
+        # Check if we have a specific site to look for
+        if "site" in request.GET:
+            # Optionally allow querying by port
+            if ":" in request.GET["site"]:
+                (hostname, port) = request.GET["site"].split(":")
```

Comment:
```suggestion
                (hostname, port) = request.GET["site"].split(":", 1)
```

Otherwise this would crash if ``:`` appears more than once